test/integration/targets/setup_docker/tasks/main.yml

```diff
@@ -30,8 +35,52 @@
       ignore_errors: yes
 
     - set_fact:
+        docker_cli_version: "{{ (docker_cli_version_stdout.stdout | default('0.0')) or '0.0' }}"
```

Comment:
Why have the `|default('0.0')` when it seems like `or '0.0'` is working for the api version example

Reply:
The difference is that if `docker version` fails, `docker_cli_version_stdout.stdout` isn't there and `docker_cli_version_stdout.stdout or '0.0'` fails. I had to change it to `docker_cli_version_stdout.stdout | default('0.0')` to stop failing (this happens for centos/6). The `or '0.0'` is there to make sure the result is not an empty string (which could happen if the command does not fail, but outputs an empty string for whatever reason).

Reply:
(The API and docker-py version checks always have `stdout` in the result, since the Python interpreter can be called. In case `docker` is not installed, like on centos/6, there's no `stdout` and `|default` is needed.)

Reply:
Ahh ok makes sense, seems like this line and the below ones could be simplified with `variable.stdout | default('0.0', True)`. The second arg means the default value is used when the value is not a True like value like an empty string or `0`.

Reply:
Ah, that's good to know. I won't change the two lines below, though, since if `stdout` isn't there, something went really bad (Python interpreter not found) and the test deserves to die :)

Reply:
Hmm, that didn't work; see [here](https://app.shippable.com/github/ansible/ansible/runs/100827/47/console) for the shippable output. Here's what I tried to do:
```.diff
-        docker_cli_version: "{{ (docker_cli_version_stdout.stdout | default('0.0')) or '0.0' }}"
+        docker_cli_version: "{{ docker_cli_version_stdout.stdout | default('0.0', true) }}"
```
I'll remove the last commit.

Reply:
Hmm ok, I tested something similar locally but obviously not enough. Thanks for trying.